spice_agent/inference/actions.py

```diff
@@ -182,6 +181,15 @@ class Inference:
     def __init__(self, spice) -> None:
         self.spice = spice
         self.device = self.spice.get_device()
+        self.pipe = None
+        self.inference_job_id = None
```

Comment:
we need to insure that this is cleared out on error and completed states. the Inference class is typically NOT recreated
```suggestion
        self.inference_job_id = None
```

Reply:
it is typically better to pass this type of variable that is not used on class instantiate to the function call itself.

callback_for_stable_diffision should probably take this as an input instead of setting it on the class

Reply:
unfortunately, the callback cannot take inference_job_id as an input since the method signature is fixed. if you are concerned that the inference class can have multiple inference job id's associated with it, I can create a new InferenceTask class that deals with inference jobs one at a time.